ampersand-view.js

```diff
@@ -140,16 +139,27 @@ assign(View.prototype, {
     // initialization logic.
     initialize: function () {},
 
-    // **render** is the core function that your view can override, its job is
-    // to populate its element (`this.el`), with the appropriate HTML.
-    render: function () {
+    // **render** is the core function that your view can override. Its job is
+    // to populate its element (`this.el`), with the appropriate HTML. __render
+    // is overriden by any user-defined render.  Each call to `render()` actually
+    // executes `__render()` && `_render()`
+    __render: function () {
         this.renderWithTemplate(this);
+        this._render();
         return this;
     },
 
-    // Remove this view by taking the element out of the DOM, and removing any
-    // applicable events listeners.
-    remove: function () {
+    // called on every `render()` call, regardless if user overrides render()
+    _render: function() {
+        this.rendered = true;
+        return this;
```

Comment:
we could additionally move things like `this.trigger('rendered')` into here, which some users might like